### PR TITLE
Sidebar

### PR DESCRIPTION
**Pull request by Fatmasiam**
Relates : #19 #24 #23

### CODE REVIEW

File: Client/src/components/App.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import React from 'react'
+import React from 'react';
+import Sidebar from './common/Sidebar/index';
+
```
Comment (AbdallahAmmar96) [inline Client/src/components/App.js]: delete the index from the path 
import Sidebar from './common/Sidebar';

File: Client/src/components/App.js [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
 export default function App() {
```
Comment (AbdallahAmmar96) [inline Client/src/components/App.js]: two empty lines 
delete line 3 or 4

File: Client/src/components/common/Sidebar/Link/index.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.sublink{
+  padding: 7px;
+  cursor: pointer;
+}
+
+.sublink:active{
+  border-left: 2px solid #eee;
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/Link/index.css]: delete white space line 5

File: Client/src/components/common/Sidebar/Link/index.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+import './index.css'
+
+export default function Link({ link }) { return (<li className='sublink'> {link} </li>) }
+
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/Link/index.js]: delete white space
and just a suggestion i think this is readable  
```javascript
export default function Link({ link }) { 
    return ( <li className='sublink'> {link} </li> )
 }
```

File: Client/src/components/common/Sidebar/Linklst/index.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.li-links{
+  padding: 7px;
+}
+
+.link{
+  list-style-type: none;
+  padding-left: 10px;
+}
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/Linklst/index.css]: add a new line at the end 
delete line 4 white space

File: Client/src/components/common/Sidebar/index.css [hunk 6]
```diff
@@ -0,0 +1,27 @@
+*{
+  padding: 0;
+  margin: 0;
+  box-sizing: border-box;
+}
+
+.sidebar{
+  width:240px;
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.css:2]: delete line 6

File: Client/src/components/common/Sidebar/Linklst/index.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Link from '../Link/index';
+import './index.css'
+const sublink = [
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/Linklst/index.js]: delete index
import Link from '../Link';

File: Client/src/components/common/Sidebar/index.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './index.css'
+import Linklst from './Linklst/index';
+const links = [[['Home'],['Course'], ['Student']]];
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js:4]: delete index
 import Linklst from './Linklst';

File: Client/src/components/common/Sidebar/Linklst/index.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  ['Add Course', 'View Course'],
+  ['Add Student ', 'View Student ', 'Edit Student']
+];
+export default function Linklst({ index, link }) {
+  return (
+    <>
+      <li className='li-links'>   <i class="fas fa-home-lg"></i> {link[0]}</li>
+      <li className='li-links'> <i class="fas fa-file-certificate"></i> {link[1]}
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/Linklst/index.js]: delete spaces in line 11

File: Client/src/components/App.js [hunk 10]
```diff
@@ -1,12 +1,12 @@
...
   return (
     <div>
-      <Mobile />
-      <Descktop />
+      <Sidebar/>
+      Hello World..!
     </div>
   )
-}
+}
```
Comment (AbdallahAmmar96) [inline Client/src/components/App.js]: add new line

File: Client/src/components/common/Sidebar/index.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function Sidebar() {
+  return (
+    <div className='sidebar'>
+      <img src='https://i.imgur.com/m6QdXqO.png' alt='AMDIEAST logo' className='img' width='200px' />
+      <h3 className='title' >AMIDEAST EL</h3>
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js:10]: no style in the js files 
 width='200px'

File: Client/src/components/common/Sidebar/index.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+      <hr className='hr' />
+      <br />
+      <ul className='linklst'>
+        {links.map((link, index) => <Linklst key={index} link={link} />)}
```
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js:14]: line 13 edit the className from  linklst to linklist

File: Client/src/components/common/Sidebar/index.js (hunk 8 above)
Comment (AbdallahAmmar96) [inline Client/src/components/common/Sidebar/index.js:4]: delete the extra [ ] line 4

**Review by AbdallahAmmar96 — CHANGES_REQUESTED** [review]

**Review by naremanhilles — APPROVED** [review]

File: Client/src/components/common/Sidebar/index.js (hunk 8 above)
Comment (Fatmasiam) [reply Client/src/components/common/Sidebar/index.js:4]: Done :)

File: Client/src/components/common/Sidebar/index.js (hunk 11 above)
Comment (Fatmasiam) [reply Client/src/components/common/Sidebar/index.js:10]: Done :)

File: Client/src/components/App.js (hunk 10 above)
Comment (Fatmasiam) [reply Client/src/components/App.js]: Done :+1:

File: Client/src/components/common/Sidebar/Linklst/index.css (hunk 5 above)
Comment (Fatmasiam) [reply Client/src/components/common/Sidebar/Linklst/index.css]: Done :+1:

**Review by AbdallahAmmar96 — APPROVED** [review]

**Comment by kannanhassouna** [comment]
First of all change the branch name  :
1 - 
```
git branch -m [old-name] [new-name]
```
2 - 
```
git push origin :[old-name] [new-name]
```
3 - 
```
git push origin -u [new-name]
```

File: Client/src/components/common/Link/index.css [hunk 13]
```diff
@@ -0,0 +1,10 @@
+.sublink{
+  font-size: 16px;
+  padding: 7px;
+  cursor: pointer;
+  margin-left: 25px;
+}
+
+.sublink:active{
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.css:8]: you don't this rule, remove it. 
our deal was the make border when I in on that link (the page). 
not when I make it active,

File: Client/src/components/common/LinkList/index.css [hunk 14]
```diff
@@ -0,0 +1,12 @@
+.li-links{
+  padding: 7px;
+}
+
+.link{
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.css:5]: you class name very popular you should make it  more specific like 

```  sidebar-li-link-list```  
``` sidebar-list```
``` sidebar-li-list ```
... etc

File: Client/src/components/common/LinkList/index.css [hunk 15]
```diff
@@ -0,0 +1,12 @@
...
+  list-style-type: none;
+  padding-left: 10px;
+}
+
+.icon{
```
Comment (kannanhassouna) [inline Client/src/components/common/LinkList/index.css:10]: the same problem.
you can get a lot of conflict with other teammates

File: Client/src/components/common/Sidebar/index.css [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  color: #eee;
+  font-family:'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
+}
+
+.img{
+  margin:50px 20px 30px 20px;
+}
+
+.title{
+text-align: center;
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.css:14]: indentation  !!!

File: Client/src/components/common/Sidebar/index.css [hunk 17]
```diff
@@ -0,0 +1,25 @@
+.sidebar{
+  width:240px;
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.css:2]: i think the sidebar width is really small, can you give it more.

Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.css:2]: indentation !!! 
you should add space between the declaration of the property and the value of it

File: Client/src/components/common/Sidebar/index.css [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+
+.img{
+  margin:50px 20px 30px 20px;
+}
+
+.title{
+text-align: center;
+}
+
+.hr{
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.css:17]: please remove the generic class name

File: Client/src/components/common/Sidebar/index.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './index.css'
+import Linklst from '../LinkList';
+const links = [[['Home'],['Course'], ['Student']]];
+
+export default function Sidebar() {
+  return (
+    <div className='sidebar'>
+      <img src='https://i.imgur.com/m6QdXqO.png' alt='AMDIEAST logo' className='img' width='200px' />
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js:9]: remove generic class name AGAIN
```suggestion
      <img src='https://i.imgur.com/m6QdXqO.png' alt='AMDIEAST logo' className='sidebar-desktop-logo' width='200px' />
```

File: Client/src/components/common/Link/index.css [hunk 20]
```diff
@@ -0,0 +1,10 @@
+.sublink{
+  font-size: 16px;
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.css:2]: i think be default it's ``` 16px ```

File: Client/src/components/common/Link/index.js [hunk 21]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import './index.css'
+
+export default function Link({ link }) {
```
Comment (kannanhassouna) [inline Client/src/components/common/Link/index.js:4]: add correct property name like ``` linkTitle or title or name```

File: Client/src/components/common/Sidebar/index.js [hunk 22]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './index.css'
+import Linklst from '../LinkList';
+const links = [[['Home'],['Course'], ['Student']]];
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js:4]:
```javascript
const links = [[['Home'],['Course'], ['Student']]];
```
for what ?

File: Client/src/components/common/Sidebar/index.js [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+
+export default function Sidebar() {
+  return (
+    <div className='sidebar'>
+      <img src='https://i.imgur.com/m6QdXqO.png' alt='AMDIEAST logo' className='img' width='200px' />
+      <h3 className='title' >AMIDEAST EL</h3>
+      <hr className='hr' />
+      <br />
+      <ul className='linklist'>
+        {links.map((link, index) => <Linklst key={index} link={link} />)}
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js:14]:
```suggestion
        {links.map((link, index) => <LinkList key={index} link={link} />)}
```

File: Client/src/components/common/Sidebar/index.js [hunk 24]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import './index.css'
+import Linklst from '../LinkList';
```
Comment (kannanhassouna) [inline Client/src/components/common/Sidebar/index.js:3]:
```suggestion
import LinkList from '../LinkList';
```

**Review by kannanhassouna — CHANGES_REQUESTED** [review]
we need more reusable components !!

**Comment by kannanhassouna** [comment]
remove package-lock.json don't forget
@Fatmasiam

File: Client/src/components/common/Link/index.css (hunk 20 above)
Comment (Fatmasiam) [reply Client/src/components/common/Link/index.css:2]: the default is ```18px```

File: Client/src/components/common/Link/index.css (hunk 13 above)
Comment (Fatmasiam) [reply Client/src/components/common/Link/index.css:8]: done

File: Client/src/components/common/LinkList/index.css (hunk 15 above)
Comment (Fatmasiam) [reply Client/src/components/common/LinkList/index.css:10]: done

File: Client/src/components/common/Sidebar/index.css (hunk 17 above)
Comment (Fatmasiam) [reply Client/src/components/common/Sidebar/index.css:2]: ```20%``` it's good ?

**Pull request closed without merging**